go/vt/vtctl/reparentutil/emergency_reparenter.go

```diff
@@ -518,13 +519,26 @@ func (erp *EmergencyReparenter) reparentReplicas(
 	rec := concurrency.AllErrorRecorder{}
 
 	handlePrimary := func(alias string, tablet *topodatapb.Tablet) error {
-		position, err := erp.tmc.PrimaryPosition(replCtx, tablet)
-		if err != nil {
-			return err
-		}
-		if populateReparentJournal {
+		if !intermediateReparent {
+			var position string
+			var err error
+			if ev.ShardInfo.PrimaryAlias == nil {
```

Comment:
Might be worth adding some nil checks here, although I think we catch panics in this path IIRC.

Reply:
I considered it, but decided against it, because before in the workflow, we call `ev.ShardInfo = *shardInfo`. When we call this function `ev`, `ev.SharInfo` have to be non-nil. I can still add the checks, but they won't help since they're guaranteed to be non-nil values.